package/MDAnalysis/analysis/hole2/helper.py

```diff
@@ -0,0 +1,556 @@
+
+logger = logging.getLogger(__name__)
+
+
+def write_simplerad2(filename="simple2.rad"):
+    """Write the built-in radii in :data:`SIMPLE2_RAD` to `filename`.
+
+    Does nothing if `filename` already exists.
+
+    Parameters
+    ----------
+    filename : str, optional
+       output file name; the default is "simple2.rad"
+
+    Returns
+    -------
+    filename : str
+       returns the name of the data file
+    """
+
+    if not os.path.exists(filename):
+        with open(filename, "w") as rad:
+            rad.write(SIMPLE2_RAD + "\n")
+        logger.debug("Created simple radii file {}".format(filename))
+    return filename
+
+
+def check_and_fix_long_filename(filename, tmpdir=os.path.curdir,
+                                max_length=70,
+                                make_symlink=True):
+    """Return a file name suitable for HOLE.
+
+    HOLE is limited to filenames <= ``max_length``. This method
+
+    1. returns `filename` if HOLE can process it
+    2. returns a relative path (see :func:`os.path.relpath`) if that shortens the
+        path sufficiently
+    3. creates a symlink to `filename` (:func:`os.symlink`) in a safe temporary
+        directory and returns the path of the symlink.
+
+    Parameters
+    ----------
+    filename : str
+        file name to be processed
+    tmpdir : str, optional
+        By default the temporary directory is created inside the current
+        directory in order to keep that path name short. This can be changed
+        with the `tmpdir` keyword (e.g. one can use "/tmp"). The default is
+        the current directory :data:`os.path.curdir`.
+
+    Returns
+    -------
+    str
+        path to the file that has a length less than
+        ``max_length``
+
+    Raises
+    ------
+    RuntimeError
+        If none of the tricks for filename shortening worked. In this case,
+        manually rename the file or recompile your version of HOLE.
+    """
+
+    if len(filename) <= max_length:
+        return filename
+
+    msg = ('HOLE will not read {} '
+           'because it has more than {} characters.')
+    logger.debug(msg.format(filename, max_length))
+
+    # try a relative path
+    new_name = os.path.relpath(filename)
+    if len(new_name) <= max_length:
+        msg = 'Using relative path: {} -> {}'
+        logger.debug(msg.format(filename, new_name))
+        return new_name
+
+    if make_symlink:
+        # shorten path by creating a symlink inside a safe temp dir
+        root, ext = os.path.splitext(filename)
+        dirname = tempfile.mkdtemp(dir=tmpdir)
+        newname = os.path.join(dirname, os.path.basename(filename))
+        if len(newname) > max_length:
+            fd, newname = tempfile.mkstemp(suffix='.pdb', dir=dirname)
+            os.close(fd)
+        os.symlink(filename, newname)
+        msg = 'Using symlink: {} -> {}'
+        logger.debug(msg.format(filename, newname))
+        return newname
+
+    msg = 'Failed to shorten filename {}'
```

Comment:
`RuntimeError` is not tested.

Reply:
I can't trigger the `RuntimeError` because the temporary file inside a temporary directory is short enough to return before then. In Python 2.7 the default names are 'tmp'+6 random characters -- can't find notes on Python 3 but they look similar. In theory the `RuntimeError` shouldn't happen, I guess.